src/resources/defi/PositionsQuery.ts

```diff
@@ -0,0 +1,438 @@
+          quantity: string;
+        }): { asset: ZerionAsset; quantity: string; native: NativeDisplay } => {
+          const nativeDisplay = convertRawAmountToNativeDisplay(
+            underlying.quantity,
+            underlying.asset.decimals,
+            underlying.asset.price?.value!,
+            currency
+          );
+          totalDeposits = add(totalDeposits, nativeDisplay.amount);
+
+          return {
+            ...underlying,
+            native: nativeDisplay,
+          };
+        }
+      );
+      return deposit as RainbowDeposit;
+    }
+  );
+
+  let totalBorrows = '0';
+
+  const parsedBorrows = position.borrows?.map(
+    (borrow: Borrow): RainbowBorrow => {
+      borrow.underlying = borrow.underlying.map(
+        (underlying: {
+          asset: ZerionAsset;
+          quantity: string;
+        }): { asset: ZerionAsset; quantity: string; native: NativeDisplay } => {
+          const nativeDisplay = convertRawAmountToNativeDisplay(
+            underlying.quantity,
+            underlying.asset.decimals,
+            underlying.asset.price?.value!,
+            currency
+          );
+          totalBorrows = add(totalBorrows, nativeDisplay.amount);
+
+          return {
+            ...underlying,
+            native: nativeDisplay,
+          };
+        }
+      );
+      return borrow as RainbowBorrow;
+    }
+  );
+
+  let totalClaimables = '0';
+  const parsedClaimables = position.claimables?.map(
+    (claim: Claimable): RainbowClaimable => {
+      const nativeDisplay = convertRawAmountToNativeDisplay(
+        claim.quantity,
+        claim.asset.decimals,
+        claim.asset.price?.value!,
+        currency
+      );
+      totalClaimables = add(totalClaimables, nativeDisplay.amount);
+      return {
+        asset: claim.asset,
+        quantity: claim.quantity,
+        native: nativeDisplay,
+      };
+    }
+  );
+
+  const positionTotals: PositionsTotals = {
+    totals: {
+      amount: subtract(add(totalDeposits, totalClaimables), totalBorrows),
+      display: convertAmountToNativeDisplay(
+        subtract(add(totalDeposits, totalClaimables), totalBorrows),
+        currency
+      ),
+    },
+    borrows: {
+      amount: totalBorrows,
+      display: convertAmountToNativeDisplay(totalBorrows, currency),
+    },
+    claimables: {
+      amount: totalClaimables,
+      display: convertAmountToNativeDisplay(totalClaimables, currency),
+    },
+    deposits: {
+      amount: totalDeposits,
+      display: convertAmountToNativeDisplay(totalDeposits, currency),
+    },
+  };
+
+  return {
+    type: position.type,
+    totals: positionTotals,
+    deposits: parsedDeposits,
+    borrows: parsedBorrows,
+    claimables: parsedClaimables,
+    // revert dapp name bs once versions are handled via backend
+    dapp: {
+      ...position.dapp,
+      name: position.type,
+      icon_url: maybeSignUri(position.dapp.icon_url) || position.dapp.icon_url,
+    },
+  };
+};
+const parsePositions = (
```

Comment:
imo move all these helpers out into a utils.ts file, just have RQ stuff in this file